TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -536,196 +578,199 @@ public class RoutingManager : AbstractGeometryObservingManager, IRoutingManager
 #endif
 
-								int currentLaneConnectionTransIndex = -1;
-								if (nextHasOutgoingConnections) {
-									// check for lane connections
-									if (nextIsConnectedWithPrev) {
-										// lane is connected with previous lane
-										if (numNextLaneConnectionTransitionDatas < MAX_NUM_TRANSITIONS) {
-											currentLaneConnectionTransIndex = numNextLaneConnectionTransitionDatas;
-											nextLaneConnectionTransitionDatas[numNextLaneConnectionTransitionDatas++].Set(nextLaneId, nextLaneIndex, LaneEndTransitionType.LaneConnection, nextSegmentId, isNextStartNodeOfNextSegment);
-										} else {
-											Log.Warning($"nextTransitionDatas overflow @ source lane {prevLaneId}, idx {prevLaneIndex} @ seg. {prevSegmentId}");
-										}
+                                int currentLaneConnectionTransIndex = -1;
+                                if (nextHasOutgoingConnections) {
+                                    // check for lane connections
+                                    if (nextIsConnectedWithPrev) {
+                                        // lane is connected with previous lane
+                                        if (numNextLaneConnectionTransitionDatas < MAX_NUM_TRANSITIONS) {
+                                            currentLaneConnectionTransIndex = numNextLaneConnectionTransitionDatas;
+                                            nextLaneConnectionTransitionDatas[numNextLaneConnectionTransitionDatas++].Set(nextLaneId, nextLaneIndex, LaneEndTransitionType.LaneConnection, nextSegmentId, isNextStartNodeOfNextSegment);
+                                        } else {
+                                            Log.Warning($"nextTransitionDatas overflow @ source lane {prevLaneId}, idx {prevLaneIndex} @ seg. {prevSegmentId}");
+                                        }
 #if DEBUGROUTING
 										if (debugFine)
 											Log._Debug($"RoutingManager.RecalculateLaneEndRoutingData({segmentId}, {laneIndex}, {laneId}, {startNode}): nextLaneId={nextLaneId}, idx={nextLaneIndex} has outgoing connections and is connected with previous lane. adding as lane connection lane.");
 #endif
-									} else {
+                                    } else {
 #if DEBUGROUTING
 										if (debugFine)
 											Log._Debug($"RoutingManager.RecalculateLaneEndRoutingData({segmentId}, {laneIndex}, {laneId}, {startNode}): nextLaneId={nextLaneId}, idx={nextLaneIndex} has outgoing connections but is NOT connected with previous lane");
 #endif
-									}
-								}
-
-								if (prevIsMergeLane && Constants.ServiceFactory.NetService.CheckLaneFlags(nextLaneId, NetLane.Flags.Merge)) {
+                                    }
+                                }
+                                bool laneInValidGroup = laneGroups[sortedLanes.IndexOf(nextLaneInfo)] == prevLaneGroup;
```

Comment:
Now lane groups of different segments are compared. This will not always yield a correct result because `m_position` has been used to order lanes -> results are not comparable